subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/bintray/BintrayReleasePlugin.java

```diff
@@ -61,7 +64,12 @@ public void execute(JavaBintrayPlugin plugin) {
                         deferredConfiguration(subproject, new Runnable() {
                             public void run() {
                                 UpdateReleaseNotesTask updateNotes = (UpdateReleaseNotesTask) project.getTasks().getByName(ReleaseNotesPlugin.UPDATE_NOTES_TASK);
-                                updateNotes.setPublicationRepository(BintrayUtil.getRepoLink(bintray));
+                                String userSpecifiedRepo = conf.getLenient().getReleaseNotes().getPublicationRepository();
+                                if (userSpecifiedRepo != null) {
```

Comment:
Why lenient config here, while regular one in GradlePortalReleasePlugin?

Reply:
Sry. I don't really get it 🙃 
That was original designed by @szczepiq.

Should I change that code to
```
(UpdateReleaseNotesTask) project.getTasks().getByName(ReleaseNotesPlugin.UPDATE_NOTES_TASK);
```
as well? 🤔

Reply:
It would be also good to create unit tests for these two cases.

Reply:
Hmm.. wait 🤔 
We have to get the `publicationRepository` from the config here.
Because we want to set it to the task :)
We can't use the task to get the `publicationRepository`. It is always `null` ...

Reply:
I'm only asking about lenient config here: 
`conf.getLenient().getReleaseNotes().getPublicationRepository()`
Why here it's lenient, and in GradlePortalReleasePlugin it's not lenient. 
In general we use lenient one when the property can be null. But you are right, it was like that in the first commit for this issue.
@szczepiq Why is it different for these two classes?

Reply:
Sure!

lenient() is documented in the Javadoc, let me know if it is unclear (or suggest an improvement / rename).

In this case, we should be using lenient() because repo is optional - we can get the value automatically from bintray extension. In the other case (gradle portal plugin), it is not optional, we cannot get the value from bintray extension because bintray plugin is not used.

Does it sound right? Good question!